routing/segment.hpp

```diff
@@ -71,12 +79,108 @@ class Segment final
   bool m_forward = false;
 };
 
+class SegmentWeight final
+{
+public:
+  SegmentWeight() = default;
+
+  constexpr explicit SegmentWeight(double weight) : m_weight(weight) {}
+  constexpr SegmentWeight(SegmentWeight const & segWeight) : m_weight(segWeight.m_weight) {}
+
+  double GetWeight() const { return m_weight; }
+
+  bool operator<(SegmentWeight const & segWeight) const
+  {
+    return m_weight < segWeight.m_weight;
+  }
+
+  bool operator==(SegmentWeight const & segWeight) const
+  {
+    return !(*this < segWeight) && !(segWeight < *this);
+  }
+
+  bool operator!=(SegmentWeight const & segWeight) const
+  {
+    return (*this < segWeight) || (segWeight < *this);
+  }
+
+  bool operator>(SegmentWeight const & segWeight) const
```

Comment:
We use it in current AStar implementation so there are two ways: limit AStar to use only <, ==, != or define > and >= here. I think second variant is better cause `if (a >= b)` may look better than `if ( !(a < b) )` when `a` and `b` are huge.